analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -571,38 +571,81 @@ public void Main<T>() where T : new()
 {
     var value = new T();
     Tag(""Value"", value);
-}
-
-private void Tag(string name, object arg) { }";
+}";
             var validator = SETestContext.CreateCSMethod(code).Validator;
             validator.ValidateContainsOperation(OperationKind.TypeParameterObjectCreation);
             validator.ValidateTag("Value", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
 
         [TestMethod]
-        public void Literal_Null_SetsNull_CS()
+        public void Literal_NullAndDefault_SetsNull_CS()
         {
             const string code = @"
-Tag(""Before"", arg);
-arg = null;
-Tag(""After"", arg);";
-            var validator = SETestContext.CreateCS(code, ", object arg").Validator;
+Tag(""BeforeObjNull"", argObjNull);
+Tag(""BeforeObjDefault"", argObjDefault);
+Tag(""BeforeInt"", argInt);
+argObjNull = null;
+argObjDefault = default;
+argInt = default;
```

Comment:
It would be nice to see some "conversions" happening here:
```cs
        object o; int i;
        o = default(Exception);
        i = default(byte);
```